src/constraint/encoding.ts

```diff
@@ -208,18 +195,35 @@ export const ENCODING_CONSTRAINTS: EncodingConstraintModel[] = [
       return true; // other channel is irrelevant to this constraint
     }
   }, {
-    name: 'dataTypeMatchesScaleType',
+    name: 'dataTypeAndFunctionMatchesScaleType',
     description: 'ScaleType must match data type',
-    properties: [Property.TYPE, Property.SCALE_TYPE],
+    properties: [Property.TYPE, Property.SCALE_TYPE, Property.TIMEUNIT, Property.BIN],
     requireAllProperties: true,
     strict: true,
     satisfy: (encQ: EncodingQuery, schema: Schema, stats: Stats, opt: QueryConfig) => {
       if (encQ.scale) {
         const scaleType = (encQ.scale as ScaleQuery).type;
         const type = encQ.type;
-        return (contains([ScaleType.ORDINAL, ScaleType.LINEAR, undefined], scaleType) && contains([Type.ORDINAL, Type.NOMINAL], type)) ||
-                (contains([ScaleType.TIME, ScaleType.UTC, ScaleType.ORDINAL, ScaleType.LINEAR, undefined], scaleType) && (type === Type.TEMPORAL)) ||
-                (contains([ScaleType.LOG, ScaleType.POW, ScaleType.SQRT, ScaleType.QUANTILE, ScaleType.QUANTIZE, ScaleType.LINEAR, undefined], scaleType) && (type === Type.QUANTITATIVE));
+
+        if (contains([Type.ORDINAL, Type.NOMINAL], type)) {
+          if (!encQ.timeUnit) {
```

Comment:
Easy gut check question:   why would you apply `timeUnit` to ordinal / nominal data?  (You won't)